sklearn/ensemble/_hist_gradient_boosting/tests/test_gradient_boosting.py

```diff
@@ -190,3 +190,29 @@ def test_zero_division_hessians(data):
     X, y = data
     gb = HistGradientBoostingClassifier(learning_rate=100, max_iter=10)
     gb.fit(X, y)
+
+
+@pytest.mark.parametrize('GradientBoosting, data', [
+    (HistGradientBoostingClassifier,
+     make_classification(n_samples=10001, random_state=0, n_features=2,
+                         n_informative=2, n_redundant=0)),
+    (HistGradientBoostingRegressor,
+     make_regression(n_samples=10001, random_state=0, n_features=2,
+                     n_informative=2))]
+)
+def test_small_trainset(GradientBoosting, data):
+    # Make sure that a small trainset has the expected length (10k samples)
+    X, y = data
+    gb = GradientBoosting(random_state=42)
+    X_small, y_small = gb._get_small_trainset(X, y, seed=42)
+    assert X_small.shape[0] == 10000
+    assert y_small.shape[0] == 10000
+
+
+def test_stratification_small_trainset():
+    # Make sure that the small trainset is stratified
+    X, y = make_classification(n_samples=20000, n_features=2,
+                               n_informative=2, n_redundant=0)
```

Comment:
y is balanced by default. I think you should update the test to use an imbalanced y.

```python
n_samples =20000
rng = np.random.RandomState(42)
X = rng.randn(n_samples).reshape(-1, 1)
y = (rng.rand(n_samples) > 0.9).astype(np.int32)
```

Reply:
Is it really important? If `stratify=None`, only the expected value of the mean of `y_train` will be equal to the mean of `y` (according to the law of large numbers), whereas if `stratify=y`, the mean of `y_train` will be equal to the mean of `y` (excluding rounding approximation -- that's why I set `decimal=3` in the test).

I don't think that changing the balance of `y` will change the tested behavior, because it is independent of the balance of `y` imo.